tests/library/browsercontext-credentials.spec.ts

```diff
@@ -64,6 +89,41 @@ it('should fail with wrong credentials', async ({ browser, server }) => {
   await context.close();
 });
 
+it('should fail with wrong credentials and matching hostname', async ({ browser, server }) => {
+  server.setAuth('/empty.html', 'user', 'pass');
+  const hostname = URL.parse(server.PREFIX).hostname;
+  const context = await browser.newContext({
+    httpCredentials: { hostname: hostname, username: 'foo', password: 'bar' }
```

Comment:
All these tests will pass today, w/o any changes that you were adding. You should assert that the credentials are not sent, in here you actually send wrong credentials and that's why your tests pass. Should you fix it, the tests will start failing. Once Chromium is fixed, you would need to implement it for Firefox and WebKit as well.

Reply:
Right, I fixed the test "should fail with correct credentials and mismatching hostname". Current code wouldn't work with this additional test case.
This test assert that the credentials are not send (because they are correct, and we still got a 401 as response).
 
Then agree the couple other test are not very relevant since it fails anyway because of wrong credentials... do you want me to remove them to keep tests cleaner ?